docs/tutorial.rst

```diff
@@ -563,3 +569,101 @@ and manually updating the action set to use artifacts created by the
 previous action set can be cumbersome. In situations like this, it is
 useful to instead use ``kanctl``. To learn how to leverage ``kanctl`` to
 create action sets, see :ref:`architecture` .
+
+Blueprint Development
+=====================
+
+To start with Blueprint Development we need to be aware of some terms such as
+Actions, Phase, KanisterFunctions, Objects which has already been covered in
+this tutorial.
+
+Define and add ``actions`` for the blueprint as per your application. Each action
+performs a set of tasks. For example, backup, restore, and delete.
+
+``actions`` in the blueprint consists of ``phases`` and each phase executes a
+:ref:`functions`
+which we specify through ``func`` field inside the phase. We can find the
+list of :ref:`existingfunctions` supported by Kanister here. Arguments to the
+functions can be provided through ``args`` field.
+
+Each phase can consume or create :ref:`artifacts`. Artifacts
+are generally used to reference backed-up data. For example, If there is
+backup action performed on S3 then the artifact
+will hold a reference to that data.
+Artifacts created by phases in an action are Output Artifacts.
+Artifacts consumed by an action are Input Artifacts.
+
+An action can also consume information from Kubernetes objects
+such as :ref:`Secret<consumingSecret>` and :ref:`ConfigMap<consumingConfigMap>`
+
+To consume data from objects we need to reference it using ``configMapNames``
+and ``secretNames`` fields in an action.
+
```

Comment:
We can also pass secrets as objects right?
```
  objects:
        pgSecret:
          kind: Secret
          name: '{{ .StatefulSet.Name }}'
          namespace: '{{ .StatefulSet.Namespace }}'
 ```

Reply:
I think we pass these objects inside `phases` and not `action`.

```
type BlueprintAction struct {
	Name               string              `json:"name"`
	Kind               string              `json:"kind"`
	ConfigMapNames     []string            `json:"configMapNames,omitempty"`
	SecretNames        []string            `json:"secretNames,omitempty"`
	InputArtifactNames []string            `json:"inputArtifactNames,omitempty"`
	OutputArtifacts    map[string]Artifact `json:"outputArtifacts,omitempty"`
	Phases             []BlueprintPhase    `json:"phases,omitempty"`
}

```

```
type BlueprintPhase struct {
	Func       string                     `json:"func"`
	Name       string                     `json:"name"`
	ObjectRefs map[string]ObjectReference `json:"objects,omitempty"`
	Args       map[string]interface{}     `json:"args"`
}
```